File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        m = {c: i for i, c in enumerate(order)}
        l = [[m[c] for c in w] for w in words]
        return not any(x > y for (x, y) in zip(l, l[1:]))

File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.cpp
class Solution {
public:
    bool isAlienSorted(const vector<string>& words, const string& order) {
        array<int, 26> ord {};
        for(int i = 0; i < order.size(); ++i) {
            ord[order[i] - 'a'] = i;
        }
        
        auto comp = [&ord] (char a, char b) {
            return ord[a - 'a'] < ord[b - 'a'];  
        };
        
        for(int i = 1; i < words.size(); ++i) {
            const auto& prev = words[i-1];
            const auto& curr = words[i];
            if (lexicographical_compare(curr.begin(), curr.end(), prev.begin(), prev.end(), comp)) {
                return false;
            }
        }
        
        return true;
    }
};